### PR TITLE
Implement Functions for Cardano Metadata APIs with Unit and Integration Tests

### PR DESCRIPTION
**Pull request by hieu-dd**
# Pull Request (PR) Summary

- This PR introduces new functions for interacting with Cardano Metadata APIs in the SDK, along with comprehensive unit and integration tests.

## Changes Description

This PR adds a set of functions to interact with the Cardano Metadata APIs, enabling the handling and retrieval of transaction metadata on the Cardano blockchain. These functions are essential for applications that require embedding or retrieving additional information with blockchain transactions.

To ensure the reliability and functionality of these new functions, I have implemented both unit tests and integration tests. These tests are designed to verify that the functions perform correctly under various scenarios and handle both typical and edge cases effectively.

## Related Issue(s)

- Fixes # (issue number)  // Please replace with actual issue number if applicable

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other (please describe)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] My PR is targeting the `main` branch
- [x] I have added necessary documentation (if appropriate)
- [x] I have added tests that prove my fix is effective or my feature works
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/metadata/model/TransactionMetadataLabel.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * @property label Metadata label
+ * @property cip10 CIP10 defined description
+ * @property count The count of metadata entries with a specific label
+ */
+@Serializable
+data class TransactionMetadataLabel(
+    val label: String,
+    val cip10: String?,
+    val count: String,
```
Comment (iTanChi) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/metadata/model/TransactionMetadataLabel.kt]: count này là số chứ nhỉ

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/metadata/model/TransactionMetadataContentCBOR.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * Transaction Metadata Content in CBOR
+ *
+ * @property txHash Transaction hash that contains the specific metadata
+ * @property cborMetadata Content of the CBOR metadata (deprecated)
+ * @property metadata Content of the CBOR metadata in hex
+ */
+@Serializable
+data class TransactionMetadataContentCBOR(
+    @SerialName("tx_hash") val txHash: String,
+    @SerialName("cbor_metadata") val cborMetadata: String?,
```
Comment (iTanChi) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/metadata/model/TransactionMetadataContentCBOR.kt:17]: thêm thông tin deprecated vào field này luôn?

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/base/JsonExtension.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package dev.kryptonreborn.blockfrost.base
+
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+
+internal fun JsonObject.toMap(): Map<String, Any> {
```
Comment (iTanChi) [inline core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/base/JsonExtension.kt]: nên đưa vào utils/json thì ok hơn, mà chỗ này có dùng poly của serialization được không nhỉ?

File: core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/metadata/model/TransactionMetadataLabel.kt (hunk 1 above)
Comment (hieu-dd) [reply core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/metadata/model/TransactionMetadataLabel.kt]: trong docs là string

Comment (hieu-dd) [reply core/src/commonMain/kotlin/dev/kryptonreborn/blockfrost/metadata/model/TransactionMetadataLabel.kt]: hay nó là biginteger?

File: core/src/commonTest/kotlin/dev/kryptonreborn/blockfrost/unittest/base/BigIntegerSerializerTest.kt [hunk 4]
```diff
@@ -1,23 +1,34 @@
...
-import dev.kryptonreborn.blockfrost.ktor.Ktor
+import dev.kryptonreborn.blockfrost.base.BigIntegerSerializer
 import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.modules.SerializersModule
 import kotlin.test.Test
 import kotlin.test.assertEquals
 
 class BigIntegerSerializerTest {
+    private val json =
```
Comment (iTanChi) [inline core/src/commonTest/kotlin/dev/kryptonreborn/blockfrost/unittest/base/BigIntegerSerializerTest.kt:12]: sao cần custom nhỉ

Comment (hieu-dd) [reply core/src/commonTest/kotlin/dev/kryptonreborn/blockfrost/unittest/base/BigIntegerSerializerTest.kt:12]: Test riêng cho thằng bigInteger thì phải thế, k dùng cái @file: kia được. Nếu k thì chỉ test được wrapper của nó thôi

**Review by iTanChi — DISMISSED** [review]

**Review by iTanChi — APPROVED** [review]
